utils/src/utils/document_plan.clj

```diff
@@ -1,60 +1,64 @@
 (ns utils.document-plan
   (:gen-class)
   (:require [api.nlg.parser :refer [document-plan->semantic-graph]]
-            [jsonista.core :as json]
+            [clojure.java.io :as io]
             [clojure.pprint :refer [pprint]]
-            [utils.queries :as queries]
-            [mount.core :refer [defstate] :as mount]
-            [utils.config :refer [config]]
             [clojure.tools.logging :as log]
-            [org.httpkit.client :as http]))
-
+            [jsonista.core :as json]
+            [mount.core :as mount]
+            [org.httpkit.client :as http]
+            [utils.config :refer [config]]
+            [utils.queries :as queries]))
 (def read-mapper (json/object-mapper {:decode-key-fn true}))
 
-(defn ->semantic-graph [file-path]
-  (-> (slurp file-path)
-      (json/read-value read-mapper)
-      :documentPlan
-      (document-plan->semantic-graph)
-      (pprint)))
+(defn pprint-semantic-graph [file-path]
+  (let [dp       (json/read-value (slurp file-path) read-mapper)
+        fixed-dp (assoc dp
+                        :documentPlan
+                        (json/read-value (:documentPlan dp) read-mapper))]
+    (-> fixed-dp
+        (document-plan->semantic-graph)
+        (pprint))))
 
-(defn ouput-document-plan [dir dp])
+(defn doc->dir-name [{kind :kind}]
+  (condp = kind
+    "RGL" "dlg" "Document" "dp" "AMR" "amr"))
 
 (defn ->file [output-dir {:keys [id] :as document-plan}]
-  (let [fpath (format "%s/%s.json" output-dir id)]
-    (log/debugf "Writing to: %s" fpath)
-    (spit
-     fpath
-     (json/write-value-as-string document-plan))))
+  (let [dir (str output-dir "/" (doc->dir-name document-plan))]
+    (.mkdirs (io/file dir))
+    (log/infof "Writing: %s/%s.json" dir id)
+    (spit (format "%s/%s.json" dir id)
+          (json/write-value-as-string document-plan))))
```

Comment:
here we can use `(with-out-str (println (json/write-value-as-string document-plan (json/object-mapper {:pretty true})))` for pretty printed string, then write